lib/SILOptimizer/Analysis/EscapeAnalysis.cpp

```diff
@@ -376,6 +376,27 @@ updatePointsTo(CGNode *InitialNode, CGNode *pointsTo) {
   clearWorkListFlags(WorkList);
 }
 
+int EscapeAnalysis::ConnectionGraph::addUsePoint(CGNode *Node, SILNode *User) {
+  // For escaping nodes we have to make worst case assumptions anyway. So no
+  // need to store use point information.
+  if (Node->getEscapeState() >= EscapeState::Global &&
+      // ... except it has defer predecessor edges, which are (potentially) not
+      // escaping: use-points are propagated in reverse direction along defer
+      // edges!
+      !Node->hasDeferPredecessors()) {
```

Comment:
This appears to be fixing an existing correctness issue. Is that right?

Reply:
No, There was not correctness issue.
Not propagating in reverse direction of defer edges was worked-around by also checking the content node - which is the same for all nodes in the defer-web.

Reply:
I see what you're saying. But if the direction of the defer edge matters, then I think phi arguments needed defer edges in both directions. Or, we just say that the direction doesn't matter and propagate both escape states and use points in both directions.

If the direction does not matter, then we could probably just merge the nodes in the defer web.